core/aws-core/src/main/java/software/amazon/awssdk/awscore/client/builder/AwsDefaultClientBuilder.java

```diff
@@ -191,7 +193,8 @@ protected final SdkClientConfiguration finalizeChildConfiguration(SdkClientConfi
         configuration = mergeSmartDefaults(configuration);
 
         return configuration.toBuilder()
-                            .option(AwsClientOption.CREDENTIALS_PROVIDER, resolveCredentials(configuration))
```

Comment:
Unfortunately we have to keep setting this field, since it's a protected API and old client versions would still rely on it being there. Another thing we can clean up later?

Reply:
That is probably sufficient? I don't think we need to set CREDENTIALS_PROVIDER before configuration finalization, but we can check the child classes to be sure. I don't think there are old versions at play for this field that we need to be defensive about.

If it's easier to just set both in the `credentialsProvider` methods than figuring out how to play it safe, I don't think it's a problem.